src/Microsoft.Net.Http.Headers/CookieHeaderParser.cs

```diff
@@ -91,7 +91,7 @@ private static int GetNextNonEmptyOrWhitespaceIndex(string input, int startIndex
 
             if (skipEmptyValues)
             {
-                while ((current < input.Length) && (input[current] == ',') && (input[current] == ';'))
+                while ((current < input.Length) && ((input[current] == ',') || (input[current] == ';')))
```

Comment:
Woah, how weird is that?! @Tratcher does this also mean that there's a bug in the original code this was borrowed from?